### ISSUE TITLE
Allow aliases (duplicate keys to same value)

### ISSUE THREAD

**Issue by pnacht** [decisionpatterns/r-hash#19]
It'd be nice if it were possible to create hash-tables with more than one key per value.

That is, if I have the hash

```
h = hash(keys = letters, values = 1:26)
```

I'd like to be able to set an alias such as the collection LETTERS which returns the same values as defined by `keys`.

This can be roughly done with

```
h = hash(keys = c(letters, LETTERS), values = rep(1:26, 2))
```

But that's unsafe (changing h['a'] won't modify h['A']), and requires twice the memory allocation (in real cases where the hash overhead is trivial).

Would it be possible to add an alias parameter to .set() (and therefore the constructor), such that `.set(keys = c(...), alias = c(...), values = c(...))` has the aliases point to the same values as the keys?

**Comment by ctbrown** [comment]
Let me think about this a bit.

On first thought.  No, I don't think that this is a great idea.  It breaks
the notion of what a hash table is, a unique key associated with a value.
I think a better solution is to wrap the key lookups with a
transformation.  Instead of h['a']  use h[ toupper('a') ].  This allows
solves your example case.  I get that it is a drag to replace each lookup
with a function call.

I slightly better approach would allow specification of a function that
would coerce the keys before performing setting and getting.  Such as set(
keys=..., values=..., key.formatter = toupper ).  Again, I am not sure this
is a great idea as it impacts overhead.

I will think about both.

**Comment by pnacht** [comment]
Yeah, the letters/LETTERS example was just an example for any generic alias format. But the solution would be the same: create a hash with the original keys mapped to the values, and a secondary hash mapping aliases to keys. Then wrap lookups in a function which converts any aliases to keys and then gets the values.
